src/lib/FreeC/Backend/Agda/Converter/Type.hs

```diff
@@ -66,10 +96,36 @@ dagger = fmap free . star
 --   > (τ₁ → τ₂)* = τ₁' → τ₂'
 --   > C* = Ĉ Shape Position
 --   > α* = α̂
-star :: IR.Type -> Converter Agda.Expr
-star (IR.TypeVar s name) = Agda.Ident
+star :: Maybe Agda.Expr -> IR.Type -> Converter Agda.Expr
+star (Just i) t                   = Agda.app <$> star' t <*> pure i
+star _        (IR.TypeVar s name) = Agda.Ident
   <$> lookupAgdaIdentOrFail s IR.TypeScope (IR.UnQual (IR.Ident name))
-star (IR.TypeCon s name) =
+star _ (IR.TypeCon s name) =
   applyFreeArgs <$> lookupAgdaIdentOrFail s IR.TypeScope name
-star (IR.TypeApp  _ l r) = Agda.app <$> star l <*> star r
-star (IR.FuncType _ l r) = Agda.fun <$> dagger l <*> dagger r
+star _ (IR.TypeApp  _ l r) = Agda.app <$> star' l <*> star' r
+star _ (IR.FuncType _ l r) = Agda.fun <$> dagger' l <*> dagger' r
+
+dagger' :: IR.Type -> Converter Agda.Expr
+dagger' = dagger Nothing
+
+star' :: IR.Type -> Converter Agda.Expr
+star' = star Nothing
```

Comment:
Add documentation for `pi`, `dagger'`, `star'` and the additional parameter of `dagger` and `star`. I'd also suggest to find better names for these functions. For example, `dagger'` could just be called `convertType` and `dagger` something like `convertSizedType`.